### PR TITLE
User experience - set env on the machine

### PR DESCRIPTION
**Pull request by sushilraje**
# Description and Motivation 

...

# Change type 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-cli/341)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -331,6 +332,8 @@ function main() {
                         answers.dockerTag = 'testing';
                     } else if (program.dockerTagOverride) {
                         answers.dockerTag = program.dockerTagOverride;
+                    } else if (program.setLocalEnvironments) {
```
Comment (ppathan) [inline src/index.ts]: I don't think this should be in else if. Should be after line 325. If it is not set then it should be undefined or null

File: src/deploymentmanager.ts [hunk 2]
```diff
@@ -212,7 +213,11 @@ export class DeploymentManager implements IDeploymentManager {
                 }
 
                 if (answers.deploymentSku === 'local') {
-                    this.printEnvironmentVariables(deploymentProperties.outputs, storageEndpointSuffix);
+                    if (answers.setLocalEnv) {
```
Comment (ppathan) [inline src/deploymentmanager.ts]: Should we not set the env on the machine by default?

File: src/deploymentmanager.ts [hunk 3]
```diff
@@ -553,6 +558,60 @@ export class DeploymentManager implements IDeploymentManager {
 
         console.log('Copy the following environment variables to /scripts/local/.env file: \n\ %s', `${chalk.cyan(data.join('\n'))}`);
     }
+
+    private setEnvironmentVariables(outputs: any, storageEndpointSuffix: string, environmentFileName: string) {
```
Comment (ppathan) [inline src/deploymentmanager.ts]: printEnvironmentVariables and setEnvironmentVariables has lot of common code. Refactor it into common method

File: src/index.ts [hunk 4]
```diff
@@ -96,6 +96,7 @@ const program = new Command(packageJson.name)
     .option('--domainId <domainId>', 'This can either be an .onmicrosoft.com domain or the Azure object ID for the tenant')
     .option('--solutionName <solutionName>', 'Solution name for your Remote monitoring accelerator')
     .option('--subscriptionId <subscriptionId>', 'SubscriptionId on which this solution should be created')
+    .option('--setLocalEnvironments <file>', 'Globally sets environment variables on the machine and saves it to afile. (Works with sku local).')
```
Comment (ppathan) [inline src/index.ts]: space between `a file`

File: src/deploymentmanager.ts (hunk 3 above)
Comment (sushilraje) [reply src/deploymentmanager.ts]: yes I was thinking over that

File: src/deploymentmanager.ts (hunk 2 above)
Comment (sushilraje) [reply src/deploymentmanager.ts]: I though giving an option would be better.

File: src/deploymentmanager.ts [hunk 5]
```diff
@@ -551,7 +552,40 @@ export class DeploymentManager implements IDeploymentManager {
...
+                    console.log('The environment could not be set. unable to determine OS.');
+                    break; 
+                 }
+            }
+            cp.exec(cmd);
+
+        });
+
+        //fs.writeFileSync(environmentFileName, data.join('\n'));
+        //console.log('The environment variables have been set on this machine. A copy of these variables is saved in the file  \n\ %s', '.env');
```
Comment (ppathan) [inline src/deploymentmanager.ts]: remove?

File: package-lock.json [hunk 6]
```diff
@@ -1048,13 +1048,13 @@
...
           "dev": true,
           "requires": {
-            "debug": "^2.3.3",
-            "define-property": "^0.2.5",
-            "extend-shallow": "^2.0.1",
-            "posix-character-classes": "^0.1.0",
-            "regex-not": "^1.0.0",
-            "snapdragon": "^0.8.1",
-            "to-regex": "^3.0.1"
+            "debug": "2.6.9",
```
Comment (ppathan) [inline package-lock.json]: I'm not sure why there is package-lock.json changes. Can you revert?

File: src/deploymentmanager.ts (hunk 5 above)
Comment (sushilraje) [reply src/deploymentmanager.ts]: Just keeping here, if we wish to add this functionality later.

File: package-lock.json (hunk 6 above)
Comment (sushilraje) [reply package-lock.json]: yes

File: src/deploymentmanager.ts (hunk 5 above)
Comment (ppathan) [reply src/deploymentmanager.ts]: I would delete the code that is commented.

**Pull request merged**